### PR TITLE
Deps bump, type/API naming, docs fixes, feature flag docs.

### PR DESCRIPTION
**Pull request by virtualritz**
* Made all names conform to [Rust API naming guidelines](https://rust-lang.github.io/api-guidelines/naming.html).
  
   This breaks the API so I bumpled version to `0.2`. I already have fixes/a PR for the `colstodian` crate (which hasn't been updated since 3 years) which is the only noteworthy of the two dependents on `kolor` listed on `crates.io`.

* Fixed docs (grammar, typos, quoting of API refs., etc.).

* Added support for auto-documenting feature flags.

* Improved UX of CAT module.

* Bumped deps (i.e. `glam`) to latest.

### CODE REVIEW

**Comment by kabergstrom** [comment]
Hey @virtualritz thanks for the cleanup!

I find the Rust naming conventions make acronyms less readable for me personally, but I am happy to merge the changes in the interest of conforming to the ecosystem.

Do you use kolor for any of your own projects?

**Comment by virtualritz** [comment]
I probably will, indirectly, though `colstodian` soon.
TLDR; working on a VFX comparing system in Rust (closed source) but some deps will be/are OSS.

**Comment by virtualritz** [comment]
> I find the Rust naming conventions make acronyms less readable for me personally, but I am happy to merge the changes in the interest of conforming to the ecosystem.

I also shared some reservations initially but got used to the rules now. And having stringent naming accross the ecosystem just wins, IMHO.
